x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/AsyncSearchIntegTestCase.java

```diff
@@ -148,40 +169,25 @@ protected void ensureTaskCompletion(String id) throws Exception {
     }
 
     protected SearchResponseIterator assertBlockingIterator(String indexName,
```

Comment:
could you add a quick comment explaining what this iterator allows to do?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/56026/commits/0304ec6b3b521e2ebe47c40a9c064797203bc021